models/group.rb

```diff
@@ -14,6 +14,24 @@ def participants
               .all
   end
 
+  def participants_by_rank
+    result = Hash.new
+    key    = 0
+    score  = 0
+
+    participants.each do |participant|
+      if score.zero? || score > (participant[:score] || 0)
+        score = participant[:score] || 0
+        key += 1
+      end
+
+      result[key] = Array.new unless result.has_key?(key)
```

Comment:
I guess you can use `result[key] ||= Array.new` here

Reply:
Updated. Thanks